tests/test_iterations.py

```diff
@@ -4,6 +4,7 @@
 
 import pytest
```

Comment:
This is unrelated to this line but the thread about the failures with custom parameters:

With this upgrade we use the botorch factory instead of instatiating ourselves the mc acqfs. This means we do not pass a precomputed `best_f` into the call. Instead, the factory uses this line to compute `best_f` itself: 
```python
        Y = model.posterior(X_observed, posterior_transform=posterior_transform).mean
        obj = objective(samples=Y, X=X_observed)
        best_f = compute_best_feasible_objective(
            samples=Y,
            obj=obj,
            constraints=constraints,
            model=model,
            objective=objective,
            posterior_transform=posterior_transform,
            X_baseline=X_observed,
        )
```

It throws an error `RuntimeError: The q-batch shape of the objective values does not agree with the q-batch shape of X. Got 1 and 2. This may happen if you used a one-to-many input transform but forgot to use a corresponding objective.`. But only for the custom examples called `_architecture_torch` and `_pretrained`. The `_architecture_sklearn` one if fine. This results in the two examples failing the fulltest.

I found that the object returned from `posterior.mean` has shape `[2,1]` in the sklearn case and shape `[2,1,1]` in the others, hence I think something with our custom surrogate wrappers etc might be at fault here. This has not surfaced since this particular line was not invoked before this PR, it seems the optimization ran fine despite this issue, this needs to be investigated further.

Reply:
resolved by not using the factory

Reply:
But still making me curious: are we doing something wrong in our construction? I mean, if botorch does it differently, perhaps there is a reason for it?

Reply:
this is what the docstring of `qExpectedImprovement` suggests:
`best_f = train_Y.max()[0]` so that looks exactly like what we do and did before

The factory does compute `best_f` like
```python
        Y = model.posterior(X_observed, posterior_transform=posterior_transform).mean
        obj = objective(samples=Y, X=X_observed)
        best_f = compute_best_feasible_objective(
            samples=Y,
            obj=obj,
            constraints=constraints,
            model=model,
            objective=objective,
            posterior_transform=posterior_transform,
            X_baseline=X_observed,
        )
```

and the utility `compute_best_feasible_objective` only does special things if its called with constraints a la 
```python
    if constraints is None:  # unconstrained case
        # we don't need to differentiate through X_baseline for now, so taking
        # the regular max over the n points to get best_f is fine
        with torch.no_grad():
            return obj.amax(dim=-1, keepdim=False)
```
so that would suggest to me were fine? Were not currently using the constrained objective mechanism

the issue I was facing comes from this line `Y = model.posterior(X_observed, posterior_transform=posterior_transform).mean` where Y doesnt seem to have the right shape for 2 of our custom surrogate examples

Reply:
Ok, great, thank you for doing the extra work and diggin' this up for me 🏅 Agree, we should be OK for now 👍🏼